File: backend/src/main/java/net/archasmiel/skufapi/dto/auth/MeResponse.java
package net.archasmiel.skufapi.dto.auth;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import net.archasmiel.skufapi.dto.IResponse;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Response with Jwt token")
public class MeResponse implements IResponse {

  @Schema(description = "Jwt token", example = "eyJhbGciOiJIUzUxMiJ9...")
  private String username;

  @Schema(description = "Jwt token", example = "eyJhbGciOiJIUzUxMiJ9...")
  private String roles;
}

File: backend/src/main/kotlin/net/archasmiel/skufapi/api/response/auth/MeResponse.kt
package net.archasmiel.skufapi.api.response.auth

import io.swagger.v3.oas.annotations.media.Schema
import net.archasmiel.skufapi.api.response.ApiResponse

@Schema(description = "Response with user details")
data class MeResponse(
    @field:Schema(description = "Username", example = "john_doe")
    val username: String,

    @field:Schema(description = "User roles", example = "ROLE_USER,ROLE_ADMIN")
    val roles: String
) : ApiResponse